src/rdkafka_admin.c

```diff
@@ -6289,16 +6300,22 @@ rd_kafka_DescribeConsumerGroupsResponse_parse(rd_kafka_op_t *rko_req,
                                               char *errstr,
                                               size_t errstr_size) {
         const int log_decode_errors = LOG_ERR;
-        int cnt, nodeid, port;
+        int nodeid, port;
+        int16_t api_version;
+        int32_t cnt;
         rd_kafka_op_t *rko_result = NULL;
         rd_kafka_broker_t *rkb    = reply->rkbuf_rkb;
         rd_kafka_Node_t *node     = NULL;
         rd_kafka_error_t *error   = NULL;
-        char *group_s = NULL, *group_state_s = NULL, *proto_type_s = NULL,
-             *proto_s = NULL, *host = NULL, *member_id_s = NULL,
-             *client_id_s = NULL, *client_host_s = NULL;
+        char *group_id = NULL, *group_state = NULL, *proto_type = NULL,
+             *proto = NULL, *host = NULL;
+
+        api_version = rd_kafka_buf_ApiVersion(reply);
+        if (api_version >= 1) {
+                rd_kafka_buf_read_throttle_time(reply);
+        }
 
-        rd_kafka_buf_read_i32(reply, &cnt);
+        rd_kafka_buf_read_arraycnt(reply, &cnt, -1);
```

Comment:
For any value that we read from the network and then use to allocate memory, we need to provide a limit check to make sure we don't crash if the value is way out of bounds.
See "Protocol level sanity" in rdkafka_int.h

For this here, I'd say probably 100k would be a safe bet.

Reply:
done